File: frontend/src/types/report.ts
export interface EmployeeReport {
  userId: string
  userName: string
  count: number
  totalAmount: number
}

export interface CategoryReport {
  categoryId: string
  categoryName: string
  count: number
  totalAmount: number
  percentage: number
}

export interface ReportSummary {
  year: number
  month: number
  totalCount: number
  grandTotal: number
}

export interface EmployeeReportResponse {
  summary: ReportSummary
  employees: EmployeeReport[]
}

export interface CategoryReportResponse {
  summary: ReportSummary
  categories: CategoryReport[]
}

File: backend/internal/model/report.go
package model

// EmployeeReport は社員別月別集計レポートの1行を表す。
type EmployeeReport struct {
	UserID      string `json:"userId"`
	UserName    string `json:"userName"`
	Count       int    `json:"count"`
	TotalAmount int    `json:"totalAmount"`
}

// CategoryReport は勘定科目別集計レポートの1行を表す。
type CategoryReport struct {
	CategoryID   string  `json:"categoryId"`
	CategoryName string  `json:"categoryName"`
	Count        int     `json:"count"`
	TotalAmount  int     `json:"totalAmount"`
	Percentage   float64 `json:"percentage"`
}

// ReportSummary は集計レポート全体のサマリーを表す。
type ReportSummary struct {
	Year       int `json:"year"`
	Month      int `json:"month"`
	TotalCount int `json:"totalCount"`
	GrandTotal int `json:"grandTotal"`
}

// EmployeeReportResponse は社員別集計レポートのAPIレスポンス。
type EmployeeReportResponse struct {
	Summary   ReportSummary    `json:"summary"`
	Employees []EmployeeReport `json:"employees"`
}

// CategoryReportResponse は勘定科目別集計レポートのAPIレスポンス。
type CategoryReportResponse struct {
	Summary    ReportSummary    `json:"summary"`
	Categories []CategoryReport `json:"categories"`
}
